railties/lib/rails/test_unit/minitest_plugin.rb

```diff
@@ -102,7 +102,34 @@ def self.plugin_rails_init(options)
     reporter << ::Rails::TestUnitReporter.new(options[:io], options)
   end
 
-  mattr_accessor(:run_via) { Hash.new }
+  @@run_via = nil unless defined? @@run_via
+
+  # For backward compatibility, makes it possible to set via `[]=`.
+  def @@run_via.[]=(runner, _)
```

Comment:
How about revising the surface to:

```ruby
mattr_reader(:run_via) { RunVia.new }

def self.run_via=(runner)
  if run_via.set?
    raise # …
  else
    run_via.runner = runner
  end
end

class RunVia
  attr_accessor :runner
  alias set? runner

  # Backwardscompatibility with Rails 5.0 generated plugin test scripts.
  alias []= runner=

  def ruby?
    @runner == :ruby
  end

  def rake?
    @runner == :rake
  end
end
```

Would also prevent `NoMethodError`'s in cases where `run_via.rake?` is called without first having something assigned (albeit an unlikely scenario).